network/stream/stream.go

```diff
@@ -497,16 +480,17 @@ func (r *Registry) serverHandleGetRange(ctx context.Context, p *Peer, msg *GetRa
 		batchSizeGauge.Update(int64(l))
 	}
 	if err := p.Send(ctx, offered); err != nil {
-		p.logger.Error("erroring sending offered hashes", "ruid", msg.Ruid, "err", err)
 		p.mtx.Lock()
 		delete(p.openOffers, msg.Ruid)
 		p.mtx.Unlock()
-		p.Drop("error sending offered hashes")
+		return fmt.Errorf("error sending offered hashes: %w", err)
```

Comment:
i chose to delete the entry for the sake of cleanliness, but indeed it is not necessary since the whole peer object should be disposed of when a send error occurs